src/index.ts

```diff
@@ -20,72 +20,135 @@
+} from "@inrupt/solid-client-authn-node";
+import express from "express";
 
-const argv = require("yargs/yargs")(process.argv.slice(2))
-  .describe(
-    "oidcIssuer",
-    "The identity provider at which the user should authenticate."
-  )
-  .alias("issuer", "oidcIssuer")
-  .describe("clientName", "The name of the bootstrapped app.")
-  .demandOption(["oidcIssuer"])
-  .locale("en")
-  .help().argv;
+import {
+  promptClientName,
+  promptIdp,
+  promptPort,
+  promptRegistration,
+  promptStaticClientInfo,
+} from "./prompts";
 
-import express from "express";
-const app = express();
-const PORT = 3001;
-const iriBase = `http://localhost:${PORT}`;
-const storage = new InMemoryStorage();
+type InputOptions = {
+  solidIdentityProvider?: string;
+  clientName?: string;
+  registrationType?: "static" | "dynamic";
+};
+
+type ValidatedOptions = {
+  solidIdentityProvider: string;
+  registrationType: "static" | "dynamic";
+  clientName?: string;
+};
 
-// Initialised when the server comes up and is running...
-let session: Session;
+async function main(): Promise<void> {
+  // Get CLI arguments
+  const argv = require("yargs/yargs")(process.argv.slice(2))
+    .describe(
+      "solidIdentityProvider",
+      "The identity provider at which the user should authenticate."
+    )
+    .alias("idp", "solidIdentityProvider")
+    .describe("appName", "The name of the app you are registering.")
+    .describe(
+      "registration",
+      "[static] if you want to manually register the client, [dynamic] otherwise."
+    )
+    .describe(
+      "port",
+      "The port number on which the identity provider will return the token."
+    )
+    .locale("en")
+    .help().argv;
 
-const server = app.listen(PORT, async () => {
-  session = new Session({
+  const inputOptions: InputOptions = {
+    ...argv,
+    clientName: argv.appName,
+  };
+  // Complete CLI arguments with user prompt
+  const validatedOptions: ValidatedOptions = {
+    solidIdentityProvider:
+      inputOptions.solidIdentityProvider ?? (await promptIdp()),
+    registrationType:
+      inputOptions.registrationType ?? (await promptRegistration()),
+    clientName: inputOptions.clientName ?? (await promptClientName()),
+  };
+  const port = argv.port ?? (await promptPort());
+
+  const app = express();
+  const iriBase = `http://localhost:${port}`;
+  const storage = new InMemoryStorage();
+
+  const session: Session = new Session({
     insecureStorage: storage,
     secureStorage: storage,
   });
 
-  console.log(`Listening at: [http://localhost:${PORT}].`);
-  console.log(`Logging in ${argv.oidcIssuer} to get a refresh token.`);
-  session
-    .login({
-      clientName: argv.clientName,
-      oidcIssuer: argv.oidcIssuer,
+  const server = app.listen(port, async () => {
+    console.log(`Listening at: [${iriBase}].`);
+    const loginOptions: ILoginInputOptions = {
+      clientName: validatedOptions.clientName,
+      oidcIssuer: validatedOptions.solidIdentityProvider,
```

Comment:
Yeah, we need to update the SCA Core interfaces to use `solidIdentityProvider`, and then update this (later I suppose...)